Stop tracking removed files and modules

szajbus opened the pull request:
After a file is removed
* its md5 should not be tracked anymore and 
* its module should not be tracked in `mod_deps`

axelson commented:
Hi, thanks for the PR! Can you explain a little bit about what problems occur without this change and why this is needed in addition to the removal that happens on line 351?

https://github.com/elixir-lsp/elixir-ls/blob/457e16d3e58a5cac00d501b74d44884554d0f0bf/apps/language_server/lib/language_server/dialyzer.ex#L351

szajbus commented:
I added some extra JsonRPC calls to get more output from ElixirLS in vs code and I noticed that a file I already removed is still being tracked (passed in `state.removed_files` to `compile/2` on each build).

The expected behaviour would be to stop tracking removed files.

This was done by removing file's md5 hash from `md5` map that is used by `ElixirLS.LanguageServer.Dialyzer` to identify changed and removed files on builds.

But there's also `mod_deps` map that is tracking module dependencies (as provided by `:dialyzer_analysis_callgraph`) so that we can reanalyze changed modules **and also** their dependants.

When a module is removed we should stop tracking it as a dependency of other modules, because it may lead to their unnecessary reanalysis.

(On a side note, I think there may still be some issue with the module deps tracking, because it sometimes reanalyzes less modules than are actually recompiled after a change. That was the reason I started "digging" in the first place, but didn't fully resolve it yet)
